src/engine.rs

```diff
@@ -558,23 +579,27 @@ impl ApplyBackend for Engine {
                         Engine::set_code(&address, &code)
                     }
 
-                    if reset_storage {
-                        Engine::remove_all_storage(&address)
-                    }
+                    // TODO: When does the storage needs to be wiped?
+                    // if reset_storage {
+                    // Engine::remove_all_storage(&address, generation)
+                    // }
```

Comment:
We need to understand what is the meaning of the flag `reset_storage`. From [this comment](https://github.com/aurora-is-near/sputnikvm/blob/master/src/backend/mod.rs#L38-L40) it seems that the storage needs to be wiped, however, as seen in some tests, this is not working as expected.

Maybe the comment in sputnikvm refers to a different storage 🤔

Reply:
I see the error. `remove_all_storage` increments the generation by 1, but then in the storage iterator we are using the previous generation. We should do the following:

```
if reset_storage {
    Engine::remove_all_storage(&address, generation);
    generation += 1;
}
```

Reply:
I confirmed all tests pass with that change, but actually, that solution is not so good because it interacts poorly with the `if delete_empty` block later. Need to think about it more carefully.

Reply:
Fixed in https://github.com/aurora-is-near/aurora-engine/pull/84/commits/dc49d44e224e5038bcf1a44bbd00c11fdbf80095